core/src/main/scala/com/databricks/labs/remorph/parsers/FunctionBuilder.scala

```diff
@@ -206,33 +221,45 @@ object FunctionBuilder {
     case "UPPER" => Some(FixedArity(1))
     case "USER_ID" => Some(VariableArity(0, 1))
     case "USER_NAME" => Some(VariableArity(0, 1))
+    case "VALUE" => Some(FixedArity(2, XmlFunction))
     case "XACT_STATE" => Some(FixedArity(0))
     case "YEAR" => Some(FixedArity(1))
     case _ => None
   }
   // GCOVR_EXCL_STOP
 
+  def functionType(name: String): FunctionType = {
+    val uName = name.toUpperCase(Locale.getDefault())
+    val defnOption = functionArity(uName)
+    defnOption match {
+      case Some(fixedArity: FixedArity) => fixedArity.functionType
+      case Some(variableArity: VariableArity) => variableArity.functionType
+      case None => StandardFunction
+    }
+  }
+
   def buildFunction(name: String, args: Seq[ir.Expression]): ir.Expression = {
     val uName = name.toUpperCase(Locale.getDefault())
     val defnOption = functionArity(uName)
 
     defnOption match {
-      case Some(FixedArity(_, false)) | Some(VariableArity(_, _, false)) =>
-        // Should raise a warning/lint error here
+      case Some(fixedArity: FixedArity) if !fixedArity.isConvertible =>
         ir.UnresolvedFunction(name, args, is_distinct = false, is_user_defined_function = false)
 
-      case Some(FixedArity(arity, true)) if args.length == arity =>
+      case Some(fixedArity: FixedArity) if args.length == fixedArity.arity =>
         ir.CallFunction(name, args)
 
-      case Some(VariableArity(argMin, argMax, true)) if args.length >= argMin && args.length <= argMax =>
+      case Some(variableArity: VariableArity) if !variableArity.isConvertible =>
+        ir.UnresolvedFunction(name, args, is_distinct = false, is_user_defined_function = false)
+
+      case Some(variableArity: VariableArity)
+          if args.length >= variableArity.argMin && args.length <= variableArity.argMax =>
         ir.CallFunction(name, args)
 
       case Some(_) =>
-        // Should raise a warning/lint error here about mismatched argument count
         ir.UnresolvedFunction(name, args, is_distinct = false, is_user_defined_function = false)
 
       case None =>
-        // Should raise a warning/lint error here about unknown function
         ir.UnresolvedFunction(name, args, is_distinct = false, is_user_defined_function = false)
```

Comment:
These two cases could be merged as a single `case _ =>`

Reply:
Except I think that down the line, we will weant to distinguish between them if we are linting, so I wanted to leave that there. If we are not ever going to lint, then they can be merged of course.